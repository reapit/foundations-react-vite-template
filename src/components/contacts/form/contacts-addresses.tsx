import { cx } from '@linaria/core'
import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
  Button,
  elMb7,
} from '@reapit/elements'
import { Dispatch, FC, SetStateAction, useState } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { COUNTRY_OPTIONS } from './contacts-country-options'
import { ContactFormSchema } from './contacts-validation-schema'

export const ADDRESS_TYPE_OPTIONS = [
  { name: 'Primary', value: 'primary' },
  { name: 'Secondary', value: 'secondary' },
  { name: 'Home', value: 'home' },
  { name: 'Work', value: 'work' },
  { name: 'Forwarding', value: 'forwarding' },
  { name: 'Company', value: 'company' },
  { name: 'Previous', value: 'previous' },
]

interface ContactsAddressesProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const handleAddAddress = (setHasAsAddress: Dispatch<SetStateAction<boolean>>, hasAddress: boolean) => () => {
  setHasAsAddress(!hasAddress)
}

export const ContactsAddresses: FC<ContactsAddressesProps> = ({ form }) => {
  const [hasSecondaryAddress, setHasSecondaryAddress] = useState<boolean>(false)
  const [hasWorkAddress, setHasWorkAddress] = useState<boolean>(false)

  const {
    register,
    formState: { errors },
  } = form

  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide up to three addresses for your contact. All addresses are optional, only primary address is shown by
          default.
        </BodyText>
      </div>
      <BodyText>Primary Address</BodyText>
      <FormLayout className={cx(elMb11, elFadeIn)}>
        <InputWrap>
          <InputGroup>
            <Select {...register('primaryAddress.type')} defaultValue="primary">
              <option key="default-option" value="">
                None selected
              </option>
              {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Address Type</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.buildingName')}
            type="text"
            label="Building Name"
            placeholder="Name of the building"
          />
          {errors.primaryAddress?.buildingName?.message && (
            <InputError message={errors.primaryAddress.buildingName.message} />
          )}
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.buildingNumber')}
            type="text"
            label="Building Number"
            placeholder="Number of the building"
          />
          {errors.primaryAddress?.buildingNumber?.message && (
            <InputError message={errors.primaryAddress.buildingNumber.message} />
          )}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line1')} type="text" label="Line 1" placeholder="Address line 1" />
          {errors.primaryAddress?.line1?.message && <InputError message={errors.primaryAddress.line1.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line2')} type="text" label="Line 2" placeholder="Address line 2" />
          {errors.primaryAddress?.line2?.message && <InputError message={errors.primaryAddress.line2.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line3')} type="text" label="Line 3" placeholder="Address line 3" />
          {errors.primaryAddress?.line3?.message && <InputError message={errors.primaryAddress.line3.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line4')} type="text" label="Line 4" placeholder="Address line 4" />
          {errors.primaryAddress?.line4?.message && <InputError message={errors.primaryAddress.line4.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.postcode')}
            type="text"
            label="Post Code"
            placeholder="Address post code"
          />
          {errors.primaryAddress?.postcode?.message && <InputError message={errors.primaryAddress.postcode.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Select {...register('primaryAddress.countryId')} defaultValue="GB">
              <option key="default-option" value="">
                None selected
              </option>
              {COUNTRY_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Country</Label>
          </InputGroup>
        </InputWrap>
      </FormLayout>
      {hasSecondaryAddress ? (
        <>
          <BodyText>Secondary Address</BodyText>
          <FormLayout className={cx(elMb11, elFadeIn)}>
            <InputWrap>
              <InputGroup>
                <Select {...register('secondaryAddress.type')} defaultValue="secondary">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Address Type</Label>
              </InputGroup>
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.buildingName')}
                type="text"
                label="Building Name"
                placeholder="Name of the building"
              />
              {errors.secondaryAddress?.buildingName?.message && (
                <InputError message={errors.secondaryAddress.buildingName.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.buildingNumber')}
                type="text"
                label="Building Number"
                placeholder="Number of the building"
              />
              {errors.secondaryAddress?.buildingNumber?.message && (
                <InputError message={errors.secondaryAddress.buildingNumber.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line1')}
                type="text"
                label="Line 1"
                placeholder="Address line 1"
              />
              {errors.secondaryAddress?.line1?.message && (
                <InputError message={errors.secondaryAddress.line1.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line2')}
                type="text"
                label="Line 2"
                placeholder="Address line 2"
              />
              {errors.secondaryAddress?.line2?.message && (
                <InputError message={errors.secondaryAddress.line2.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line3')}
                type="text"
                label="Line 3"
                placeholder="Address line 3"
              />
              {errors.secondaryAddress?.line3?.message && (
                <InputError message={errors.secondaryAddress.line3.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line4')}
                type="text"
                label="Line 4"
                placeholder="Address line 4"
              />
              {errors.secondaryAddress?.line4?.message && (
                <InputError message={errors.secondaryAddress.line4.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.postcode')}
                type="text"
                label="Post Code"
                placeholder="Address post code"
              />
              {errors.secondaryAddress?.postcode?.message && (
                <InputError message={errors.secondaryAddress.postcode.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup>
                <Select {...register('secondaryAddress.countryId')} defaultValue="GB">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {COUNTRY_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Country</Label>
              </InputGroup>
            </InputWrap>
          </FormLayout>
          <div className={elMb7}>
            <Button onClick={handleAddAddress(setHasSecondaryAddress, hasSecondaryAddress)} intent="low">
              Remove Secondary Address
            </Button>
          </div>
        </>
      ) : (
        <div className={elMb7}>
          <Button onClick={handleAddAddress(setHasSecondaryAddress, hasSecondaryAddress)} intent="low">
            Add Secondary Address
          </Button>
        </div>
      )}
      {hasWorkAddress ? (
        <>
          <BodyText>Work Address</BodyText>
          <FormLayout className={cx(elMb11, elFadeIn)}>
            <InputWrap>
              <InputGroup>
                <Select {...register('workAddress.type')} defaultValue="work">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Address Type</Label>
              </InputGroup>
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.buildingName')}
                type="text"
                label="Building Name"
                placeholder="Name of the building"
              />
              {errors.workAddress?.buildingName?.message && (
                <InputError message={errors.workAddress.buildingName.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.buildingNumber')}
                type="text"
                label="Building Number"
                placeholder="Number of the building"
              />
              {errors.workAddress?.buildingNumber?.message && (
                <InputError message={errors.workAddress.buildingNumber.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line1')} type="text" label="Line 1" placeholder="Address line 1" />
              {errors.workAddress?.line1?.message && <InputError message={errors.workAddress.line1.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line2')} type="text" label="Line 2" placeholder="Address line 2" />
              {errors.workAddress?.line2?.message && <InputError message={errors.workAddress.line2.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line3')} type="text" label="Line 3" placeholder="Address line 3" />
              {errors.workAddress?.line3?.message && <InputError message={errors.workAddress.line3.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line4')} type="text" label="Line 4" placeholder="Address line 4" />
              {errors.workAddress?.line4?.message && <InputError message={errors.workAddress.line4.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.postcode')}
                type="text"
                label="Post Code"
                placeholder="Address post code"
              />
              {errors.workAddress?.postcode?.message && <InputError message={errors.workAddress.postcode.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup>
                <Select {...register('workAddress.countryId')} defaultValue="GB">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {COUNTRY_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Country</Label>
              </InputGroup>
            </InputWrap>
          </FormLayout>
          <div className={elMb7}>
            <Button onClick={handleAddAddress(setHasWorkAddress, hasWorkAddress)} intent="low">
              Remove Work Address
            </Button>
          </div>
        </>
      ) : (
        <div className={elMb7}>
          <Button onClick={handleAddAddress(setHasWorkAddress, hasWorkAddress)} intent="low">
            Add Work Address
          </Button>
        </div>
      )}
    </>
  )
}
