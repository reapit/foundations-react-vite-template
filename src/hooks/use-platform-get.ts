import { useEffect, useMemo } from 'react'
import {
  StringMap,
  getMergedHeaders,
  handleReapitError,
  getUrl,
  RC_SESSION_MISSING_ERROR,
  NETWORK_ERROR,
} from './utils'
import { useSnack } from '@reapit/elements'
import { useQuery, useQueryClient } from '@tanstack/react-query'
import axios, { AxiosError } from 'axios'
import { useNavigate } from 'react-router'
import { reapitConnectBrowserSession } from '../core/connect-session'

export type PlatformGet<DataType> = [
  data: DataType | null,
  loading: boolean,
  error: string | null,
  refresh: (queryParams?: Object) => void,
  refreshing: boolean,
  clearCache: () => void,
]

export interface UsePlatformGetParams {
  path: string
  errorMessage?: string
  successMessage?: string
  queryParams?: Object
  headers?: StringMap
  fetchWhenTrue?: any[]
  retry?: number | boolean
  onSuccess?: (message: string) => void
  onError?: (message: string) => void
}

export const usePlatformGet = <DataType>({
  path,
  errorMessage,
  successMessage,
  queryParams,
  headers,
  fetchWhenTrue,
  onSuccess,
  onError,
  retry,
}: UsePlatformGetParams): PlatformGet<DataType> => {
  const { success: successSnack, error: errorSnack } = useSnack()
  const navigate = useNavigate()
  const queryClient = useQueryClient()

  const isEnabled = fetchWhenTrue?.length
    ? Boolean(fetchWhenTrue?.filter(Boolean).length === fetchWhenTrue.length)
    : true

  const url = useMemo(getUrl(path, queryParams), [path, queryParams])

  const { data, error, isLoading, isSuccess, isError, isRefetching, refetch } = useQuery<DataType, AxiosError<any>>({
    queryKey: [url],
    queryFn: async () => {
      const reqHeaders = await getMergedHeaders(reapitConnectBrowserSession, headers)

      if (!reqHeaders) throw new Error(RC_SESSION_MISSING_ERROR)

      const req = await axios.get<DataType>(url, {
        headers: reqHeaders,
      })
      return req.data
    },
    retry: retry === undefined ? 1 : retry,
    refetchOnWindowFocus: false,
    enabled: isEnabled,
  })

  useEffect(() => {
    if (isSuccess) {
      if (onSuccess && successMessage) onSuccess(successMessage)
      if (!onSuccess && successMessage) successSnack(successMessage)
    }
  }, [isSuccess, successMessage, onSuccess, successSnack])

  useEffect(() => {
    if (isError) {
      const isRcError = error.message === RC_SESSION_MISSING_ERROR
      const isFourOOne = error.code === NETWORK_ERROR
      if (isRcError || isFourOOne) {
        return navigate('/login')
      }

      const errorString = handleReapitError(error, errorMessage)
      if (onError && errorString) onError(errorString)
      if (!onError && errorString) errorSnack(errorString)
      console.error(errorString)
    }
  }, [isError, error, errorMessage, onError, errorSnack, navigate])

  const result = data ? data : null
  const errorString = error?.message ? error.message : null
  const loading = isEnabled && isLoading
  const clearCache = () => {
    queryClient.removeQueries({ queryKey: [url] })
  }

  return [result, loading, errorString, refetch, isRefetching, clearCache]
}
