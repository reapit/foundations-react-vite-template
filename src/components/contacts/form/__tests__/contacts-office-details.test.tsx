import { ContactFormSchema } from '../contacts-validation-schema'
import { handleAddAddress } from '../contacts-addresses'
import { UseFormReturn } from 'react-hook-form'
import { ContactsOfficeDetails } from '../contacts-office-details'
import { usePlatformGet } from '../../../../hooks'
import {
  mockSourceModelPagedResult,
  mockContactCategories,
  mockNegotiatorModelPagedResult,
  mockOfficeModelPagedResult,
  mockContactModel,
} from '../../../../scripts/stubs'
import { render } from '../../../../scripts/tests'

jest.mock('../../../../hooks', () => ({
  usePlatformGet: jest.fn(),
  usePlatformUpdate: jest.fn(() => [jest.fn()]),
}))

const mockUsePlatformGet = usePlatformGet as jest.Mock

const error = {
  message: 'Error',
}

describe('ContactsOfficeDetails', () => {
  it('should match a snapshot', () => {
    mockUsePlatformGet
      .mockReturnValue([mockSourceModelPagedResult])
      .mockReturnValue([mockContactCategories])
      .mockReturnValue([mockNegotiatorModelPagedResult])
      .mockReturnValue([mockOfficeModelPagedResult])

    const form = {
      register: jest.fn(() => ({})),
      getValues: jest.fn(() => 'MOCK_ID'),
      formState: {
        errors: {
          source: error,
          negotiatorIds: error,
          officeIds: error,
          categoryIds: error,
        },
      },
    } as unknown as UseFormReturn<ContactFormSchema, any>

    expect(render(<ContactsOfficeDetails form={form} contact={mockContactModel} />)).toMatchSnapshot()
  })
})

describe('handleAddAddress', () => {
  it('should submit a contact then navigate', () => {
    const setHasAsAddress = jest.fn()
    const hadAsAddress = true

    const curried = handleAddAddress(setHasAsAddress, hadAsAddress)

    curried()

    expect(setHasAsAddress).toHaveBeenCalledWith(false)
  })
})
