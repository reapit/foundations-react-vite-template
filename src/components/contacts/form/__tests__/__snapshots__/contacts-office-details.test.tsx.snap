// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsOfficeDetails should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-97 el-snack-holder"
      />
      <p
        class="el-text-base el-has-bold-text el-has-section-margin el-body-text"
      >
        Provide information here relating to your Reapit organisation & office.
      </p>
      <div
        class="el-fade-in el-form-layout"
      >
        <div
          class="mocked-styled-139 el-input-wrap-full"
        >
          <div
            class="el-mb3 el-input-group"
          >
            <input
              class="mocked-styled-38 el-input"
              id="test-static-id"
              placeholder="Search"
            />
            <span
              class="el-icon-new el-icon"
              role="img"
              style="font-size: 1rem;"
            >
              <svg
                fill="none"
                height="1em"
                style="font-size: 1rem;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                  fill="currentColor"
                />
              </svg>
            </span>
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Source
            </label>
          </div>
          <div
            aria-controls=":r0:"
            aria-haspopup="listbox"
            class="mocked-styled-58 el-multi-select-input-wrapper"
            role="combobox"
          >
            <input
              class="mocked-styled-57 el-multi-select-input"
              id="source-select"
            />
            <div
              class="mocked-styled-55 el-multi-select-selected"
            />
            <div
              class="mocked-styled-56 el-multi-select-un-selected"
            >
              <input
                class="mocked-styled-52 el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="el-has-grey-chips el-multi-select-label"
                for="test-static-id"
              >
                London Estate Agent
              </label>
              <input
                class="mocked-styled-52 el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="el-has-grey-chips el-multi-select-label"
                for="test-static-id"
              >
                Propertyfinder WhatsApp
              </label>
            </div>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-139 el-input-wrap-full"
        >
          <div
            class="el-mb3 el-input-group"
          >
            <input
              class="mocked-styled-38 el-input"
              id="test-static-id"
              placeholder="Search"
            />
            <span
              class="el-icon-new el-icon"
              role="img"
              style="font-size: 1rem;"
            >
              <svg
                fill="none"
                height="1em"
                style="font-size: 1rem;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                  fill="currentColor"
                />
              </svg>
            </span>
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Negotiators
            </label>
          </div>
          <div
            aria-controls=":r1:"
            aria-haspopup="listbox"
            class="mocked-styled-58 el-multi-select-input-wrapper"
            role="combobox"
          >
            <input
              class="mocked-styled-57 el-multi-select-input"
              id="negotiators-select"
            />
            <div
              class="mocked-styled-55 el-multi-select-selected"
            />
            <div
              class="mocked-styled-56 el-multi-select-un-selected"
            >
              <input
                class="mocked-styled-52 el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="el-has-grey-chips el-multi-select-label"
                for="test-static-id"
              >
                Mr Alfred Williamson
              </label>
              <input
                class="mocked-styled-52 el-multi-select-checkbox"
                id="test-static-id"
                type="checkbox"
              />
              <label
                class="el-has-grey-chips el-multi-select-label"
                for="test-static-id"
              >
                London Estate Agent
              </label>
            </div>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-139 el-input-wrap-full"
        >
          <div
            class="el-mb3 el-input-group"
          >
            <input
              class="mocked-styled-38 el-input"
              id="test-static-id"
              placeholder="Search"
            />
            <span
              class="el-icon-new el-icon"
              role="img"
              style="font-size: 1rem;"
            >
              <svg
                fill="none"
                height="1em"
                style="font-size: 1rem;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                  fill="currentColor"
                />
              </svg>
            </span>
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Offices
            </label>
          </div>
          <div
            aria-controls=":r2:"
            aria-haspopup="listbox"
            class="mocked-styled-58 el-multi-select-input-wrapper"
            role="combobox"
          >
            <input
              class="mocked-styled-57 el-multi-select-input"
              id="offices-select"
            />
            <div
              class="mocked-styled-55 el-multi-select-selected"
            />
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-97 el-snack-holder"
    />
    <p
      class="el-text-base el-has-bold-text el-has-section-margin el-body-text"
    >
      Provide information here relating to your Reapit organisation & office.
    </p>
    <div
      class="el-fade-in el-form-layout"
    >
      <div
        class="mocked-styled-139 el-input-wrap-full"
      >
        <div
          class="el-mb3 el-input-group"
        >
          <input
            class="mocked-styled-38 el-input"
            id="test-static-id"
            placeholder="Search"
          />
          <span
            class="el-icon-new el-icon"
            role="img"
            style="font-size: 1rem;"
          >
            <svg
              fill="none"
              height="1em"
              style="font-size: 1rem;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                fill="currentColor"
              />
            </svg>
          </span>
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Source
          </label>
        </div>
        <div
          aria-controls=":r0:"
          aria-haspopup="listbox"
          class="mocked-styled-58 el-multi-select-input-wrapper"
          role="combobox"
        >
          <input
            class="mocked-styled-57 el-multi-select-input"
            id="source-select"
          />
          <div
            class="mocked-styled-55 el-multi-select-selected"
          />
          <div
            class="mocked-styled-56 el-multi-select-un-selected"
          >
            <input
              class="mocked-styled-52 el-multi-select-checkbox"
              id="test-static-id"
              type="checkbox"
            />
            <label
              class="el-has-grey-chips el-multi-select-label"
              for="test-static-id"
            >
              London Estate Agent
            </label>
            <input
              class="mocked-styled-52 el-multi-select-checkbox"
              id="test-static-id"
              type="checkbox"
            />
            <label
              class="el-has-grey-chips el-multi-select-label"
              for="test-static-id"
            >
              Propertyfinder WhatsApp
            </label>
          </div>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-139 el-input-wrap-full"
      >
        <div
          class="el-mb3 el-input-group"
        >
          <input
            class="mocked-styled-38 el-input"
            id="test-static-id"
            placeholder="Search"
          />
          <span
            class="el-icon-new el-icon"
            role="img"
            style="font-size: 1rem;"
          >
            <svg
              fill="none"
              height="1em"
              style="font-size: 1rem;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                fill="currentColor"
              />
            </svg>
          </span>
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Negotiators
          </label>
        </div>
        <div
          aria-controls=":r1:"
          aria-haspopup="listbox"
          class="mocked-styled-58 el-multi-select-input-wrapper"
          role="combobox"
        >
          <input
            class="mocked-styled-57 el-multi-select-input"
            id="negotiators-select"
          />
          <div
            class="mocked-styled-55 el-multi-select-selected"
          />
          <div
            class="mocked-styled-56 el-multi-select-un-selected"
          >
            <input
              class="mocked-styled-52 el-multi-select-checkbox"
              id="test-static-id"
              type="checkbox"
            />
            <label
              class="el-has-grey-chips el-multi-select-label"
              for="test-static-id"
            >
              Mr Alfred Williamson
            </label>
            <input
              class="mocked-styled-52 el-multi-select-checkbox"
              id="test-static-id"
              type="checkbox"
            />
            <label
              class="el-has-grey-chips el-multi-select-label"
              for="test-static-id"
            >
              London Estate Agent
            </label>
          </div>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-139 el-input-wrap-full"
      >
        <div
          class="el-mb3 el-input-group"
        >
          <input
            class="mocked-styled-38 el-input"
            id="test-static-id"
            placeholder="Search"
          />
          <span
            class="el-icon-new el-icon"
            role="img"
            style="font-size: 1rem;"
          >
            <svg
              fill="none"
              height="1em"
              style="font-size: 1rem;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m21.71 20.29-5.4-5.39A7.9 7.9 0 0 0 18 10a8 8 0 1 0-8 8 7.9 7.9 0 0 0 4.9-1.69l5.39 5.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42M4 10a6 6 0 1 1 12 0 6 6 0 0 1-12 0"
                fill="currentColor"
              />
            </svg>
          </span>
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Offices
          </label>
        </div>
        <div
          aria-controls=":r2:"
          aria-haspopup="listbox"
          class="mocked-styled-58 el-multi-select-input-wrapper"
          role="combobox"
        >
          <input
            class="mocked-styled-57 el-multi-select-input"
            id="offices-select"
          />
          <div
            class="mocked-styled-55 el-multi-select-selected"
          />
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
